pkg/awsutils/awsutils.go

```diff
@@ -740,8 +741,40 @@ func (cache *EC2InstanceMetadataCache) awsGetFreeDeviceNumber() (int, error) {
 
 // AllocENI creates an ENI and attaches it to the instance
 // returns: newly created ENI ID
-func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnet string, numIPs int) (string, error) {
-	eniID, err := cache.createENI(useCustomCfg, sg, subnet, numIPs)
+func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnets []*string, numIPs int) (string, error) {
+	describeSubnetInput := &ec2.DescribeSubnetsInput{
+		SubnetIds: subnets,
+	}
+
+	start := time.Now()
+	result, err := cache.ec2SVC.DescribeSubnetsWithContext(context.Background(), describeSubnetInput)
+	ec2ApiReq.WithLabelValues("DescribeSubnets").Inc()
+	awsAPILatency.WithLabelValues("ModifyNetworkInterfaceAttribute", fmt.Sprint(err != nil), awsReqStatus(err)).Observe(msSince(start))
+	if err != nil {
+		checkAPIErrorAndBroadcastEvent(err, "ec2:DescribeSubnets")
+		awsAPIErrInc("DescribeSubnets", err)
+		ec2ApiErr.WithLabelValues("DescribeSubnets").Inc()
+		return "", errors.Wrap(err, "AllocENI: unable to describe subnets")
+	}
+
+	sort.SliceStable(result.Subnets, func(i, j int) bool {
```

Comment:
nit: Can you add a comment here that we sort subnets by available IP address count before determining the subnet to use?